src/bblocks_data_importers/who/ghed.py

```diff
@@ -0,0 +1,290 @@
+
+    get the data and metadata:
+    >>> data = ghed.get_data()
+    >>> metadata = ghed.get_metadata()
+
+    Export the raw data to disk:
+    >>> ghed.export_raw_data(path = "some_path")
+
+    You can clear the cached data using the `clear_cache method`. Once the cache is cleared, next time you call
+    `get_data` or `get_metadata`, the importer will download the data again.
+    >>> ghed.clear_cache()
+
+
+    Optionally you can pass a path to a local file containing the data to avoid downloading the data from the WHO.
+    >>> ghed = GHED(data_file="path_to_file")
+    If you do this, the importer will read the data from the file instead of downloading it from the WHO.
+    """
+
+    def __init__(self, data_file: str | None = None):
+        self._raw_data: io.BytesIO | None = None
+        self._data: pd.DataFrame | None = None
+        self._metadata: pd.DataFrame | None = None
+
+        self._data_file = Path(data_file) if data_file else None
+
+        # if the data file is passed and the filed does not exist, raise an error
+        if self._data_file and not self._data_file.exists():
+            raise FileNotFoundError(
+                f"The file path `{self._data_file}` does not exist. Please provide a valid file "
+                f"path."
+            )
+
+    @staticmethod
+    def _extract_raw_data() -> io.BytesIO:
+        """Extract the raw data from the GHED database
+
+        Returns:
+            A BytesIO object containing the raw data from the GHED database
+        """
+
+        try:
+            response = requests.get(URL)
+            response.raise_for_status()  # Raises an error for HTTP codes 4xx/5xx
+            return io.BytesIO(response.content)
+
+        except RequestException as e:
+            raise DataExtractionError(f"Error extracting data: {e}")
+
+    @staticmethod
+    def _read_local_data(path: Path) -> io.BytesIO:
+        """Read the data from a local file
+
+        Args:
+            path: Path to the file to read
+
+        Returns:
+            A BytesIO object containing the data from the file
+        """
+
+        try:
+            with path.open("rb") as f:
+                return io.BytesIO(f.read())
+        except Exception as e:
+            raise DataExtractionError(
+                f"Error reading data from file {path}: {e}"
+            ) from e
+
+    def __format_main_data(self) -> pd.DataFrame:
+        """
+        Format the main data from the GHED database
+
+        Returns:
+            A DataFrame with the formatted data
+        """
+        return (
+            pd.read_excel(self._raw_data, sheet_name="Data", dtype_backend="pyarrow")
+            .drop(columns=["region", "income"])
+            .melt(id_vars=["country", "code", "year"], var_name="indicator_code")
+            .rename(columns={"country": "country_name", "code": "iso3_code"})
+            .pipe(ensure_pyarrow_dtypes)
+        )
+
+    def __format_codes(self) -> pd.DataFrame:
```

Comment:
It is probably right to remove the static method but it caused the tests to fail which need to be reformatted

Reply:
Sorry I didn't update them :(